tests/test_edge_cases.py

```diff
@@ -1313,7 +1313,6 @@ class Model(BaseModel):
     assert m.model_dump() == {'sm': {'self': 123}}
 
 
-@pytest.mark.xfail(reason='need to detect and error if you override __init__; need to suggest a migration path')
```

Comment:
There's a big fat TODO comment in this test that can be deleted, unfortunately I can't suggest it because it's a bit too far down